lib/std/start.zig

```diff
@@ -392,6 +392,20 @@ fn posixCallMainAndExit() callconv(.C) noreturn {
             break :init @as([*]elf.Phdr, @ptrFromInt(at_phdr))[0..at_phnum];
         };
 
+        // Locate and retrieve the page size from the aux vector.
+        // This is necessary for the TLS area to be initialized.
+        // Aborts when the aux vector isn't found.
+        const page_size = blk: {
```

Comment:
It seems unnecessary to attempt to get the page_size here (which may fail) if the value is only used later in the comptime (!builtin.single_threaded) block. Maybe move it inside that one.

(Maybe set the cached page_size in heap.zig too from this value while you got it. That way you save having to fetch it twice later on.)

Reply:
Yeah that is leftover code from when I was using the fixed buffer for the TLS code. I'll remove it.